python/lib/evaluations/dmod/evaluations/transformation.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python3
+import typing
+import abc
+
+import pandas
+
+from . import specification
+from . import util
+
+
+class TransformationFunction(typing.Callable[[pandas.DataFrame, str, typing.Optional[str]], pandas.DataFrame]):
+    """
+    A function used to transform values in a data frame into a whole other datatype, series,
+    or anything else necessary for processing
+    """
+    @abc.abstractmethod
+    def __call__(
+            self,
+            initial_frame: pandas.DataFrame,
+            observation_key: str,
+            prediction_key: str = None,
+            *args,
+            **kwargs
+    ):
+        ...
+
+
+class ReindexTransformation(TransformationFunction):
```

Comment:
I may have missed it, but I don't see this actually being used anywhere (or the superclass above, except for here) ...

Reply:
Turns out the `TransformationFunction` stuff ended up getting replaced by the `ApplicationRules` in the data model - I can just yank this out. Good find.